public/app/percona/shared/components/PageSwitcherCard/PageSwitcherCard.types.ts

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export interface PageSwitcherProps {
```

Comment:
There is a possibility to make props generic:

```
export interface PageSwitcherProps<T> {
  values: Array<PageSwitcherValue<T>>;
  className?: string;
}
```

Then the usage in the component would be:

```
export const PageSwitcherCard = <T>({ values, className }: PageSwitcherProps<T>) => {
```

If using `React.FC` it is a bit more code to declare that:
```
const Component = <T>(props: Props<T>): React.ReactNode => {...}
```

Reply:
Cool tip thanks 👍